src/zepben/evolve/streaming/grpc/connect.py

```diff
@@ -29,21 +29,26 @@ def _secure_grpc_channel_builder(host: str = "localhost", rpc_port: int = 50051)
         .make_secure()
 
 
-def _grpc_channel_builder_from_password(client_id: str, username: str, password: str, host: str, rpc_port: int) -> GrpcChannelBuilder:
-    # noinspection PyArgumentList
-    token_fetcher = ZepbenTokenFetcher(
-        audience="https://evolve-ewb/",
-        issuer_domain="zepben.au.auth0.com",
-        auth_method=AuthMethod.AUTH0,
-        verify_certificate=True,
-        token_request_data={
-            'client_id': client_id,
-            'username': username,
-            'password': password,
-            'grant_type': 'password',
-            'scope': 'offline_access'
-        }
-    )
+def _grpc_channel_builder_from_password(client_id: str, username: str, password: str, host: str, rpc_port: int,
+                                        conf_address: Optional[str]) -> GrpcChannelBuilder:
```

Comment:
this can only be optional if we allow another way to provide audience, issuer_domain, and auth_method.

I suggest this:
```python
def _grpc_channel_builder_from_password(client_id: str, username: str, password: str, host: str, rpc_port: int,                                                                    
                                                                      conf_address: Optional[str] = None, **kwargs) -> GrpcChannelBuilder:
```

And then in your else block:

```python
token_fetcher = ZepbenTokenFetcher(**kwargs)
```

You will then have to add field/error checking in ZepbenTokenFetcher to ensure that all the arguments have been passed (audience, issuer_domain, and auth_method) and handle the failure here in a user friendly way (with a nice explicit error message). I'll note here that we should probably change verify_certificate to default to true in ZepbenTokenFetcher (i'll create a separate ticket).

Reply:
Am i crazy for wanting overloading in a dynamically typed language? perhaps.